### PR TITLE
Add filter in split_fragment_screen.py

### PR DESCRIPTION
**Pull request by SiyangPeng**
## Description
Recently found that there are multiple ligands that are not able to be recombined with the original protein. There were no error messages but the combined file of both the ligand and the protein only contains the protein. Adding a function that would run after the split to try to catch the ligands that were not able to be combined with the original protein and print out error messages if that is the case so could determine what to do with the protein and the ligand better. 

## Status
- [ ] Ready to go

## Developers certificate of origin
- [X] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

### CODE REVIEW

**Comment by SiyangPeng** [comment]
Hello, some of the split ligands are having some trouble being put back into the protein, but I do not currently know how to fix that. So I am putting on a quick bandaid of printing out the files with fragments that are having the issues so could back-track and find those fragments to see what is going on.

**Comment by SiyangPeng** [comment]
> @SiyangPeng the diff here includes 150+ test and work files along with the changes to your proposed code. You need to unstage / delete these from the diff.
> 
> In general it is best to **never** work inside the source tree of the repo itself, instead make a folder outside of the `asapdiscovery` repo and work from there. This is to clearly separate work from code and avoid a situation like this. This is fine, but can be very bad if working with confidential data.

Sorry about that. I will delete those and get a better work station.

**Comment by hmacdope** [comment]
> > @SiyangPeng the diff here includes 150+ test and work files along with the changes to your proposed code. You need to unstage / delete these from the diff.
> 
> > 
> 
> > In general it is best to **never** work inside the source tree of the repo itself, instead make a folder outside of the `asapdiscovery` repo and work from there. This is to clearly separate work from code and avoid a situation like this. This is fine, but can be very bad if working with confidential data.
> 
> 
> 
> Sorry about that. I will delete those and get a better work station. 

All good, it's no stress 😎

File: asapdiscovery-data/asapdiscovery/data/scripts/split_fragment_screen.py [hunk 1]
```diff
@@ -142,16 +144,34 @@ def split_ligands(directory, input_file, structure_name):
         output_file = str(lig_pdb_file.stem) + "_protein.pdb"
         save_openeye_pdb(lig_protein, output_path / output_file)
         # Delete the ligand pdb since no longer useful
-        os.remove(lig_pdb_file)
+        # os.remove(lig_pdb_file)
```
Comment (apayne97) [inline asapdiscovery-data/asapdiscovery/data/scripts/split_fragment_screen.py:147]: You could make this a flag (--save-intermediates). or uncomment.

**Review by apayne97 — CHANGES_REQUESTED** [review]
Minor edit, also @hmacdope is it overkill to add the zika fragment structure to AWS and add a test for this script? I think it would make sense to do that if we actually think we might use this again

**Review by apayne97 — COMMENTED** [review]
I think lines 117-123 is where you can use `Ligand.from_oemol()`, keep only non-redundant ligands, and then write out separate sdf files

**Review by apayne97 — APPROVED** [review]
LGTM!

File: asapdiscovery-data/asapdiscovery/data/scripts/split_fragment_screen.py (hunk 1 above)
Comment (hmacdope) [reply asapdiscovery-data/asapdiscovery/data/scripts/split_fragment_screen.py]: Happy to push testing to another PR, lets go ahead here in meantime?

**Review by hmacdope — APPROVED** [review]

**Pull request merged**